dog.md

```diff
@@ -0,0 +1,204 @@
+         <dog-wards> WARDS => WARDS -Set SetItem(ADDR) </dog-wards>
+```
+
+Dog Data
+--------
+
+-   `DogIlk` tracks parameters needed for `bite`ing CDPs.
+
+    -   `clip`: Liquidator
+    -   `chop`: Liquidation Penalty
+    -   `hole`: Max DAI needed to cover debt+fees of active auctions per ilk
+    -   `dirt`: Amt DAI needed to cover debt+fees of active auctions per ilk
+
+```k
+    syntax DogIlk ::= Ilk ( clip: Address, chop: Wad, hole: Rad, dirt: Rad ) [klabel(#DogIlk), symbol]
+ // ---------------------------------------------------------------------------------------
+```
+
+Dog Events
+----------
+
+```k
+    syntax CustomEvent ::= Bark(ilk: String, urn: Address, ink: Wad, art: Wad, due: FInt, clip: Address, Id: Int) [klabel(Bark), symbol]
+ //-----------------------------------------------------------------------------------------------------------------------------------
+
+    syntax DogStep ::= "emitBark" String Address Wad Wad FInt Address Int
+ // --------------------------------------------------------
+    rule <k> emitBark ILK URN INK ART DUE CLIP ID => ID ... </k>
+         <return-value> ID:Int </return-value>
+         <frame-events> ... (.List => ListItem(Bark(ILK, URN, INK, ART, DUE, CLIP, ID))) </frame-events>
+```
+
+File-able Fields
+----------------
+
+These parameters are controlled by governance:
+
+-   `vow`:  Debt Engine
+-   `hole`: Max DAI needed to cover debt+fees of active auctions
+-   `chop`: Liquidation Penalty
+-   `clip`: Liquidator
+
+```k
+    syntax DogAuthStep ::= "file" DogFile
+ // -------------------------------------
+
+    syntax DogFile ::= "vow" Address
+                     | "Hole" Rad
+                     | "hole" String Rad
+                     | "chop" String Wad
+                     | "clip" String Address
+
+ // -------------------------------------
+    rule <k> Dog . file vow VOW => . ... </k>
+         <dog-vow> _ => VOW </dog-vow>
+
+    rule <k> Dog . file Hole HOLE => . ... </k>
+         <dog-hole> _ => HOLE </dog-hole>
+
+    rule <k> Dog . file hole ILK HOLE => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... hole: (_ => HOLE) ) ... </dog-ilks>
+
+    rule <k> Dog . file chop ILK CHOP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... chop: (_ => CHOP) ) ... </dog-ilks>
+
+    rule <k> Dog . file clip ILK CLIP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: (_ => CLIP) ) ... </dog-ilks>
+```
+
+Dog Semantics
+-------------
+
+```k
+   syntax DogStep ::= "bark" String Address Address
+ // -------------------------
+   rule <k> Dog . bark ILK URN KPR
+   => #fun(ROOM
+   => #fun(DART_INITIAL
+   => #fun(DART_FINAL
+   => #fun(DINK
+   => #fun(DUE
+   => #fun(TAB
+   => call DOG_VAT . grab ILK URN CLIP DOG_VOW (wad(0) -Wad DINK) (wad(0) -Wad DART_FINAL)
+   ~> call DOG_VOW . fess DUE
+   ~> call CLIP . kick TAB DINK URN KPR
+   ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0   
+   )( DUE *Wad CHOP )
+   )( DART_FINAL *Wad Ray2Wad(RATE) )
```

Comment:
Similar comment here about precision loss potentially being a monkey wrench--`DUE` is a `rad` in the code and will carry extra precision.